crates/rethnet_eth/src/remote/methods.rs

```diff
@@ -349,9 +378,78 @@ pub struct GetLogsInput {
     pub address: Address,
 }
 
+/// Invalid percentile value error
+#[derive(thiserror::Error, Debug)]
+pub enum PercentileError {
+    /// Out of bounds value error
+    #[error("Percentile must be between 0 and 100 inclusive, instead it is {0}")]
+    OutOfBounds(f64),
+}
+
+/// The `rewardPercentiles` argument for `eth_feeHistory`.
+#[derive(Clone, Debug, PartialEq, Eq, Hash, serde::Deserialize, serde::Serialize)]
+#[repr(transparent)]
+#[serde(transparent)]
+pub struct Percentiles(Vec<Percentile>);
+
+impl TryFrom<Vec<f64>> for Percentiles {
+    type Error = PercentileError;
+
+    fn try_from(value: Vec<f64>) -> Result<Self, Self::Error> {
+        let percentiles = value
+            .into_iter()
+            .map(Percentile::try_from)
+            .collect::<Result<Vec<_>, _>>()?;
+        Ok(Self(percentiles))
+    }
+}
+
+/// A reward percentile value for `eth_feeHistory`. It's a floating point value in the range [0,
+/// 100] as per the Ethereum docs: https://ethereum.github.io/execution-apis/api-documentation/
+#[derive(Clone, Copy, Debug, serde::Deserialize, serde::Serialize)]
```

Comment:
Deriving `Deserialize` here is a mistake as it'd allow constructing an instance bypassing the range check. Will fix this